test/server/cards/01_SOR/events/Recruit.spec.js

```diff
@@ -0,0 +1,114 @@
+            describe('when there is one valid option,', function () {
+                beforeEach(function () {
+                    this.setupTest({
+                        phase: 'action',
+                        player1: {
+                            hand: ['recruit'],
+                            deck: ['viper-probe-droid', 'confiscate', 'i-am-your-father', 'surprise-strike', 'vanquish', 'cell-block-guard', 'tie-advanced'],
+                        },
+                    });
+                });
+
+                it('should prompt to choose a unit from the top 5 cards, reveal it, draw it, and move the rest to the bottom of the deck', function() {
+                    this.player1.clickCard(this.recruit);
+                    expect(this.player1).toHavePrompt('Select a card to reveal');
+                    expect(this.player1).toHaveDisabledPromptButtons([this.confiscate.title, this.iAmYourFather.title, this.surpriseStrike.title, this.vanquish.title]);
+                    expect(this.player1).toHaveEnabledPromptButtons([this.viperProbeDroid.title, 'Take nothing']);
+
+                    this.player1.clickPrompt(this.viperProbeDroid.title);
+                    expect(this.getChatLogs(2)).toContain('player1 takes Viper Probe Droid');
+                    expect(this.viperProbeDroid).toBeInLocation('hand');
+
+                    expect(this.confiscate).toBeInBottomOfDeck(this.player1, 4);
+                    expect(this.iAmYourFather).toBeInBottomOfDeck(this.player1, 4);
+                    expect(this.surpriseStrike).toBeInBottomOfDeck(this.player1, 4);
+                    expect(this.vanquish).toBeInBottomOfDeck(this.player1, 4);
+                });
+
+                it('should be allowed to choose nothing and place all cards on the bottom of the deck', function () {
+                    this.player1.clickCard(this.recruit);
+                    this.player1.clickPrompt('Take nothing');
+
+                    expect([this.viperProbeDroid, this.confiscate, this.iAmYourFather, this.surpriseStrike, this.vanquish]).toAllBeInBottomOfDeck(this.player1, 5);
+                });
+
+                it('should allow selection when deck has less than five cards', function() {
+                    this.player1.setDeck([this.viperProbeDroid, this.confiscate, this.iAmYourFather]);
+                    this.player1.clickCard(this.recruit);
+                    expect(this.player1).toHaveEnabledPromptButtons([this.viperProbeDroid.title, 'Take nothing']);
+                    expect(this.player1).toHaveDisabledPromptButtons([this.confiscate.title, this.iAmYourFather.title]);
+                    this.player1.clickPrompt(this.viperProbeDroid.title);
+
+                    expect(this.player1.deck.length).toBe(2);
+                    expect([this.confiscate, this.iAmYourFather]).toAllBeInBottomOfDeck(this.player1, 2);
+                });
+
+
+                it('when the deck is empty', function() {
```

Comment:
Small note on the naming - jasmine constructs tests by concatenating the hierarchical names together. So this test would end up being named `Recruit - integration - when there is one valid option, when the deck is empty`.

I would recommend just making this case its own describe block with its own setup just for clarity. Each individual case has to go through the whole game initialization process anyway, so there's no impact on performance or functionality.

Reply:
Looking further, probably a lot of these names could be slightly adjusted to make sense when concatenated together

Reply:
Done in https://github.com/SWU-Karabast/forceteki/pull/81/commits/213827b45e2721ddcf2442a0abfd0774405a6ea6